### PR TITLE
test(markdown-utils): test helper functions

### PR DESCRIPTION
**Pull request by christianmemije**
### What's included?
 
- Tests markdown utilities 

#### Test Steps

- [ ] `npm run test`

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

### CODE REVIEW

File: src/platform/markdown/markdown-utils/markdown-utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    TestBed.configureTestingModule({
+      declarations: [JumpToAnchorTestComponent],
+    }).compileComponents();
+  }));
+
+  it('removeLeadingHash should remove leading hashes', () => {
+    expect(removeLeadingHash('')).toBe('');
+    expect(removeLeadingHash('#')).toBe('');
+    expect(removeLeadingHash('#anchor')).toBe('anchor');
+    expect(removeLeadingHash('##anchor')).toBe('anchor');
```
Comment (thomaslennan) [inline src/platform/markdown/markdown-utils/markdown-utils.spec.ts]: Maybe add an item like `expect(removeLeadingHash('#before#anchor')).toBe('anchor');` to match what you did for trailing anchor.

File: src/platform/markdown/markdown-utils/markdown-utils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    expect(rawGithubHref(DEVELOP_RAW_HREF_WITH_ANCHOR)).toBe(DEVELOP_RAW_HREF_WITH_ANCHOR);
+  });
+
+  it('isGithubHref should check whether an href is from github.com', () => {
+    expect(isGithubHref('')).toBe(false);
+    expect(isGithubHref(undefined)).toBe(false);
+    expect(isGithubHref('http://github.com')).toBe(true);
+    expect(isGithubHref('https://github.com')).toBe(true);
+    expect(isGithubHref('https://github.com/something')).toBe(true);
+    expect(isGithubHref('github.com')).toBe(false);
```
Comment (thomaslennan) [inline src/platform/markdown/markdown-utils/markdown-utils.spec.ts]: What about something like 'http://not.github.com' or 'http://notgithub.com' or 'http://githubcom.com' Any value in testing those strings?

File: src/platform/markdown/markdown-utils/markdown-utils.ts [hunk 3]
```diff
@@ -2,64 +2,78 @@ export function removeLeadingHash(str: string): string {
...
+  return '';
 }
 
-export function normalizeAnchor(str: string): string {
+export function genHeadingId(str: string): string {
   if (str) {
-    return removeLeadingHash(str.replace(/(_|-|\s)+/g, '')).toLowerCase();
+    return removeLeadingHash(
+      str
+        .replace(/(_|-|\s)+/g, '')
```
Comment (thomaslennan) [inline src/platform/markdown/markdown-utils/markdown-utils.ts]: Comment would be nice here explaining what the REs are doing.

**Comment by christianmemije** [comment]
Let's not merge this until https://github.com/Teradata/covalent/pull/1477 goes in.

File: src/platform/markdown/markdown-utils/markdown-utils.spec.ts (hunk 2 above)
Comment (christianmemije) [reply src/platform/markdown/markdown-utils/markdown-utils.spec.ts]: I'll add some of these.

**Pull request merged**